crates/integration-tests/data/extensions/crates/authorization-010/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+use grafbase_sdk::{
+    AuthorizationExtension, Error, IntoQueryAuthorization, SubgraphHeaders, Token,
+    types::{AuthorizationDecisions, Configuration, ErrorResponse, QueryElements, ResponseElements},
+};
+
+#[derive(AuthorizationExtension)]
+struct CustomAuthorization;
+
+#[derive(serde::Deserialize)]
+struct DeniedIdsArgs {
+    ids: Vec<u32>,
+}
+
+#[derive(rkyv::Archive, rkyv::Serialize, rkyv::Deserialize, Debug)]
+struct State {
+    denied_ids: Vec<DeniedIds>,
+}
+
+#[derive(rkyv::Archive, rkyv::Serialize, rkyv::Deserialize, Debug)]
+struct DeniedIds {
+    query_element_id: u32,
+    denied_ids: Vec<u32>,
+}
+
+#[serde_with::serde_as]
+#[derive(serde::Deserialize)]
+struct Object<'a> {
+    #[serde(borrow, flatten)]
+    #[serde_as(as = "serde_with::Map<_, _>")]
+    fields: Vec<(&'a str, serde_json::Value)>,
+}
+
+impl Object<'_> {
+    fn id_as_u32(&self) -> u32 {
+        println!("{:#?}", self.fields);
+        self.fields.first().and_then(|(_, value)| value.as_u64()).unwrap() as u32
+    }
+}
+
+impl AuthorizationExtension for CustomAuthorization {
+    fn new(_: Configuration) -> Result<Self, Error> {
+        Ok(Self)
+    }
+
+    fn authorize_query(
+        &mut self,
+        headers: &mut SubgraphHeaders,
+        token: Token,
+        elements: QueryElements<'_>,
+    ) -> Result<impl IntoQueryAuthorization, ErrorResponse> {
+        let mut state = State { denied_ids: Vec::new() };
+        let mut builder = AuthorizationDecisions::deny_some_builder();
+        let error_id = builder.push_error(Error::new("Not authorized, query auth SDK010"));
+
+        for (name, elements) in elements.iter_grouped_by_directive_name() {
+            println!("{name}");
```

Comment:
Do you want to keep the `println!`?

Reply:
ah no not at all, think I've removed it in a further PR but not sure anymore. will double check, thanks!